src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,197 @@
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(getAllCarsByDriverQuery)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            CarDaoImpl carDao = new CarDaoImpl();
+            while (resultSet.next()) {
+                Optional<Car> car = carDao.get(resultSet.getLong("car_id"));
+                car.ifPresent(cars::add);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get cars by driver id " + driverId, e);
+        }
+        return cars;
+    }
+
+    private void updateCarsDriversTable(Car car) {
+        deleteDrivers(car);
+        insertDrivers(car);
+    }
+
+    private void deleteDrivers(Car car) {
+        String dropDriversByCarIdQuery = "DELETE FROM cars_drivers WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(dropDriversByCarIdQuery)) {
+            statement.setLong(1, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't drop drivers of car "
+                    + car + " in cars table cars_drivers", e);
+        }
+    }
+
+    private void insertDrivers(Car car) {
+        String insertDriverQuery = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(insertDriverQuery)) {
+            for (Driver driver : car.getDrivers()) {
+                statement.setLong(1, car.getId());
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't insert drivers to car. ", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteCarQuery = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(deleteCarQuery)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car by id: " + id, e);
+        }
+    }
+
+    private Car getCarWithManufacturer(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("id", Long.class);
+        String model = resultSet.getString("model");
+        ManufacturerDaoImpl manufacturerDao = new ManufacturerDaoImpl();
```

Comment:
again don't need to use any dao implementations in this class, if you want to get manufacturer data use modify queries of this class